src/plugins/hetero/src/plugin.cpp

```diff
@@ -0,0 +1,289 @@
+    
+    DeviceMetaInformationMap metaDevices = GetDevicePlugins(fallbackDevicesStr, fullConfig.GetDeviceConfig());
+
+    std::map<std::string, ov::SupportedOpsMap> queryResults;
+    for (auto&& metaDevice : metaDevices) {
+        const auto& deviceName = metaDevice.first;
+        const auto& device_config = metaDevice.second;
+        queryResults[deviceName] = get_core()->query_model(model, deviceName, device_config);
+    }
+
+    //  WARNING: Here is devices with user set priority
+    auto fallbackDevices = ov::DeviceIDParser::get_hetero_devices(fallbackDevicesStr);
+
+    ov::SupportedOpsMap res;
+    for (auto&& deviceName : fallbackDevices) {
+        for (auto&& layerQueryResult : queryResults[deviceName]) {
+            res.emplace(layerQueryResult);
+        }
+    }
+
+    return res;
+}
+
+void ov::hetero::Plugin::set_property(const ov::AnyMap& properties) {
+    m_cfg = Configuration{properties, m_cfg};
+}
+
+ov::Any ov::hetero::Plugin::get_property(const std::string& name, const ov::AnyMap& arguments) const {
+    const auto& add_ro_properties = [](const std::string& name, std::vector<ov::PropertyName>& properties) {
+        properties.emplace_back(ov::PropertyName{name, ov::PropertyMutability::RO});
+    };
+
+    const auto& default_ro_properties = []() {
+        std::vector<ov::PropertyName> ro_properties{ov::supported_properties,
+                                                    ov::device::full_name,
+                                                    ov::device::capabilities,
+                                                    ov::caching_properties};
+                                                    //ov::available_devices,
+                                                    //ov::device::architecture,
+                                                    //ov::range_for_async_infer_requests
+        return ro_properties;
+    };
+    const auto& default_rw_properties = []() {
+        std::vector<ov::PropertyName> rw_properties{ov::device::priorities};
+                                                    //ov::device::id,
+                                                    //ov::enable_profiling,
+                                                    //ov::hint::performance_mode,
+                                                    //ov::exclusive_async_requests,
+        return rw_properties;
+    };
+    const auto& to_string_vector = [](const std::vector<ov::PropertyName>& properties) {
+        std::vector<std::string> ret;
+        for (const auto& property : properties) {
+            ret.emplace_back(property);
+        }
+        return ret;
+    };
+    if (METRIC_KEY(SUPPORTED_METRICS) == name) {
+        auto metrics = default_ro_properties();
+
+        add_ro_properties(METRIC_KEY(SUPPORTED_METRICS), metrics);
+        add_ro_properties(METRIC_KEY(SUPPORTED_CONFIG_KEYS), metrics);
+        add_ro_properties(METRIC_KEY(IMPORT_EXPORT_SUPPORT), metrics);
```

Comment:
Why some options were removed (`ov::device::full_name`, `ov::caching_properties`, `ov::device::capabilities`)?

Reply:
They were moved to `default_ro_properties()`, and it will be combined with `SUPPORTED_METRICS` etc.